cohorts/load.py

```diff
@@ -1037,3 +1124,33 @@ def filter_not_null(df, col):
     if updated_len < original_len:
         print("Missing %s for %d patients: from %d to %d" % (col, original_len - updated_len, original_len, updated_len))
     return df
+
+def _provenance_str(provenance):
+    """ utility function used by _compare_provenance to print diff
+    """
+    return ["%s==%s" % (key, value) for (key, value) in provenance]
+
+
+def _compare_provenance(provenance, comparison):
```

Comment:
That `load_from_cache` section has a customized warning that includes the patient id, which i'm not sure is necessary here .. i'll add that option this function so that we can reuse it.